tests/test_async.py

```diff
@@ -24,6 +24,20 @@ async def test_wait_during_not_done(sleeper):
         sleeper.sleep_for_a_bit())
 
 
+@pytest.mark.asyncio
+async def test_wait_until_done_with_delay(sleeper):
+    await asyncio.gather(
+        wait_until_awake_with_delay(sleeper),
+        sleeper.sleep_for_a_bit())
+
+
+@pytest.mark.asyncio
+async def test_wait_until_done_fail_on_timeout(sleeper):
```

Comment:
if it's waiting it wont wait until done, so you can do something like test_wait_fail_on_timeout